sdk/src/wallet/update.rs

```diff
@@ -66,9 +69,20 @@ where
                     wallet_data.unspent_outputs.remove(&output_id);
                     // Update spent data fields
                     if let Some(output_data) = wallet_data.outputs.get_mut(&output_id) {
-                        // TODO https://github.com/iotaledger/iota-sdk/issues/1718
-                        // output_data.metadata.set_spent(true);
-                        output_data.is_spent = true;
+                        if !output_data.is_spent() {
+                            log::warn!(
+                                "[SYNC] Setting output {} as spent without having the OutputConsumptionMetadata",
+                                output_id
+                            );
+                            // Set 0 values because we don't have the actual metadata and also couldn't get it, probably
+                            // because it got pruned.
+                            output_data.metadata.spent = Some(OutputConsumptionMetadata::new(
+                                0.into(),
+                                TransactionId::new([0u8; TransactionId::LENGTH]),
+                                None,
```

Comment:
Do you think these placeholder values could lead to anything wrong?

Reply:
Well if it's used to fetch more information the tx that consumed it, it would fail, but there is no case where it can't fail if we don't get this data